client/my-sites/woocommerce/dashboard/required-plugins-install-view.jsx

```diff
@@ -152,6 +153,41 @@ class RequiredPluginsInstallView extends Component {
 		}
 	};
 
+	createTimeoutTimer = () => {
+		if ( this.timeoutTimer ) {
+			return;
+		}
+
+		// Timeout setup after 25 seconds.
+		this.timeoutTimer = window.setTimeout( this.timeoutElapsed, 25000 );
```

Comment:
A couple of thoughts...

(1) We should base this timeout on the `transferStatusesToTimes`, rather than introduce another hardcoded value.

(2) It would be good to reset this timeout timer whenever we detect progress. That way, if things are proceeding slowly, we won't accidentally prevent things from completing.

Reply:
Great idea! I've added resetting the timeout with these progress milestones in f57ebe7:
1. Waiting for plugin list - 41 seconds
2. Waiting plugin data - 41 seconds
3. Installing a plugin - 15 seconds
4. Activating a plugin - 15 seconds